packages/banner/src/Banner.tsx

```diff
@@ -0,0 +1,234 @@
+    background-color: ${uiColors.yellow.light3};
+
+    &:before {
+      background-color: ${uiColors.yellow.base};
+    }
+  `,
+  [Variant.Danger]: css`
+    color: ${uiColors.red.dark2};
+    border-color: ${uiColors.red.light2};
+    border-left-color: ${uiColors.red.base};
+    background-color: ${uiColors.red.light3};
+
+    &:before {
+      background-color: ${uiColors.red.base};
+    }
+  `,
+
+  [Variant.Success]: css`
+    color: ${uiColors.green.dark2};
+    border-color: ${uiColors.green.light2};
+    border-left-color: ${uiColors.green.base};
+    background-color: ${uiColors.green.light3};
+
+    &:before {
+      background-color: ${uiColors.green.base};
+    }
+  `,
+} as const;
+
+const map = {
+  [Variant.Info]: { color: uiColors.blue.base, icon: InfoWithCircleIcon },
+  [Variant.Warning]: {
+    color: uiColors.yellow.dark2,
+    icon: ImportantWithCircleIcon,
+  },
+  [Variant.Danger]: { color: uiColors.red.base, icon: WarningIcon },
+  [Variant.Success]: {
+    color: uiColors.green.base,
+    icon: CheckmarkWithCircleIcon,
+  },
+};
+
+const getTextStyle = (image: boolean, dismissible: boolean) => {
+  const styleObj: {
+    marginLeft: string | undefined;
+    marginRight: string | undefined;
+  } = {
+    marginLeft: undefined,
+    marginRight: undefined,
+  };
+
+  if (image) {
+    styleObj.marginLeft = '10px';
+
+    if (dismissible) {
+      styleObj.marginRight = '20px';
+    }
+  } else {
+    styleObj.marginLeft = '28px';
+
+    if (dismissible) {
+      styleObj.marginRight = '20px';
+    } else {
+      styleObj.marginRight = '10px';
+    }
+  }
+
+  return css`
+    align-self: center;
+    flex-grow: 1;
+    margin-left: ${styleObj.marginLeft};
+    margin-right: ${styleObj.marginRight};
+  `;
+};
+
+interface BannerProps extends React.ComponentPropsWithoutRef<'div'> {
+  /**
+   * Sets the variant for the Banner
+   *
+   * Default: `'info'`
+   */
+  variant?: Variant;
+
+  /**
+   * Illustration that will replace default Icon when the prop is supplied
+   */
+  image?: React.ReactElement;
+
+  /**
+   * Determines whether or not the Banner is dismissible
+   *
+   * Default: `false`
+   */
+  dismissible?: boolean;
+
+  /**
+   * Callback fired when dismiss button is clicked
+   *
+   * Default: `() => {}`
+   */
+  onClose?: React.MouseEventHandler;
+}
+
+/**
+ * # Banner
+ *
+ * Banner component
+ *
+ * ```
+<Banner>Banner content goes here.</Banner>
+```
+ * @param props.variant Sets the variant for the Banner.
+ * @param props.image Illustration that will replace default Icon when the prop is supplied.
+ * @param props.dismissible Determines whether or not the Banner is dismissible.
+ * @param props.onClose Callback fired when dismiss button is clicked.
+ */
+export default function Banner({
+  variant = Variant.Info,
+  dismissible = false,
+  onClose = () => {},
+  image,
+  children,
+  className,
+  ...rest
+}: BannerProps) {
+  const Icon = map[variant].icon;
```

Comment:
```suggestion
  const { icon: Icon, color } = map[variant];
```

And then update down below.

Reply:
👍